src/react-sliding-pane.tsx

```diff
@@ -131,6 +137,23 @@ export function ReactSlidingPane({
   );
 }
 
+function getDefaultWidth(from: Props["from"]): string {
+  return from === "left" || from === "right" ? "80%" : "100%";
+}
+
+function getDefaultHeight(from: Props["from"]): string {
+  return from === "bottom" ? "80%" : "100%";
+}
+
+function calcMarginTop(height: string): string {
+  const isPercentageValue = /%/.test(height);
+  if (isPercentageValue) {
+    const heightInDecimalValue = parseFloat(height.split("%")[0]) / 100;
+    return `calc(100vh - (${heightInDecimalValue} * 100vh))`;
```

Comment:
Note: Ideally, we could simply do sth like `calc(100vh - ${height})`. Unfortunately, for margin-top, percentage values refer to the `width` (not the height) of the parent. So I had to manually calculate the percentage relative to the total view height.